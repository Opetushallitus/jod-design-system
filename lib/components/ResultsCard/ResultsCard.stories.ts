import type { Meta, StoryObj } from '@storybook/react';

import { ResultsCard } from './ResultsCard';

const meta = {
  title: 'Cards/ResultsCard',
  component: ResultsCard,
  tags: ['autodocs'],
} satisfies Meta<typeof ResultsCard>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  parameters: {
    design: {
      type: 'figma',
      url: 'https://www.figma.com/file/6M2LrpSCcB0thlFDaQAI2J/cx_jod_client?node-id=585%3A15053',
    },
    docs: {
      description: {
        story: 'This is a card component for displaying a text with a label.',
      },
    },
  },
  args: {
    value: 534,
    label: 'ammatteja',
  },
};

export const Longer: Story = {
  parameters: {
    design: {
      type: 'figma',
      url: 'https://www.figma.com/file/6M2LrpSCcB0thlFDaQAI2J/cx_jod_client?node-id=585%3A15053',
    },
    docs: {
      description: {
        story: 'This is a card component for displaying a longer text with a label.',
      },
    },
  },
  args: {
    value: 2336,
    label: 'koulutuksia',
  },
};
