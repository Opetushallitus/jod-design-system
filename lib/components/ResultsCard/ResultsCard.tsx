export interface ResultsCardProps {
  /** Value shown on the card */
  value: string | number;
  /** Label shown on the card */
  label: string;
}

/**
 * This is a card component for displaying a value with a label.
 */
export const ResultsCard = ({ value, label }: ResultsCardProps) => {
  return (
    <div className="ds:flex ds:w-min ds:flex-col-reverse ds:items-center ds:rounded-[10px] ds:bg-bg-gray ds:px-8 ds:py-5 ds:font-bold">
      <span className="ds:text-heading-4 ds:text-primary-gray">{label}</span>
      <span className="ds:text-[80px] ds:leading-[110%] ds:text-primary-gray">{value}</span>
    </div>
  );
};
