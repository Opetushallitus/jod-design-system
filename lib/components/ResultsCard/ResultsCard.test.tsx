import { describe, expect, it } from 'vitest';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/vitest';

import { ResultsCard } from './ResultsCard';

describe('ResultsCard', () => {
  const text = 'Lorem ipsum';
  const label = 'Dolor sit amet';

  it('renders the text and the label', () => {
    const { container } = render(<ResultsCard value={text} label={label} />);
    expect(screen.getByText(text)).toBeInTheDocument();
    expect(screen.getByText(label)).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
