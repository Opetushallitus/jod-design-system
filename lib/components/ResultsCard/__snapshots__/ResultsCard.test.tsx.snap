// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ResultsCard > renders the text and the label 1`] = `
<div
  class="ds:flex ds:w-min ds:flex-col-reverse ds:items-center ds:rounded-[10px] ds:bg-bg-gray ds:px-8 ds:py-5 ds:font-bold"
>
  <span
    class="ds:text-heading-4 ds:text-black"
  >
    Dolor sit amet
  </span>
  <span
    class="ds:text-[80px] ds:leading-[110%] ds:text-black"
  >
    Lorem ipsum
  </span>
</div>
`;
